src/main/scala/org/bitcoins/core/gen/TransactionGenerators.scala

```diff
@@ -116,24 +124,143 @@ trait TransactionGenerators {
+    val randomNum = (scala.util.Random.nextInt() % 2).abs
+    if (randomNum == 0) unspendableCLTVTransaction else spendableCLTVTransaction
+  }
+
+  /**
+    * Creates a [[ECPrivateKey]], then creates a [[CLTVScriptPubKey]] from that private key
+    * Finally creates a [[Transaction]] that CANNNOT spend the [[CLTVScriptPubKey]] because the LockTime requirement
+    * is not satisfied (i.e. the transaction's lockTime has not surpassed the CLTV value in the [[CLTVScriptPubKey]])
+    * @return
+    */
+  def unspendableCLTVTransaction : Gen[(TransactionSignatureComponent, Seq[ECPrivateKey], ScriptNumber)] = ( for {
+    txLockTime <- NumberGenerator.uInt32s
+    //Generate script Numbers that are greater than txLockTime values. the suchThat condition is for thoroughness as
+    //a random generated ScriptNumber will almost certainly be a greater value than a random generated UInt32.
+    cltvLockTime <- NumberGenerator.uInt32s.suchThat(num => num > txLockTime).map(x => ScriptNumber(x.underlying))
+    sequence <- NumberGenerator.uInt32s.suchThat(num => num != UInt32.max)
+    (signedScriptSig, cltvScriptPubkey, privateKeys) <- ScriptGenerators.signedCLTVScriptSignature(cltvLockTime, txLockTime, sequence)
+    (creditingTx, outputIndex) = buildCreditingTransaction(cltvScriptPubkey)
+    (spendingTx, inputIndex) = buildSpendingTransaction(TransactionConstants.version,creditingTx, signedScriptSig, outputIndex, txLockTime, sequence)
+
+    txSigComponent = TransactionSignatureComponent(spendingTx, inputIndex, cltvScriptPubkey, Policy.standardScriptVerifyFlags)
+  } yield (txSigComponent, privateKeys, cltvLockTime)).suchThat(cltvLockTimesOfSameType)
+
+  /**
+    *  Creates a [[ECPrivateKey]], then creates a [[CLTVScriptPubKey]] from that private key
+    *  Finally creates a [[Transaction]] that can successfully spend the [[CLTVScriptPubKey]]
+    * @return
+    */
+  def spendableCLTVTransaction : Gen[(TransactionSignatureComponent, Seq[ECPrivateKey], ScriptNumber)] = ( for {
+    txLockTime <- NumberGenerator.uInt32s
+    //Generate UInt32 values that are less than txLockTime values. UInt32 values are then mapped to ScriptNumbers
+    cltvLockTime <- NumberGenerator.uInt32s.suchThat(num => num < txLockTime).map(x => ScriptNumber(x.underlying))
+    sequence <- NumberGenerator.uInt32s.suchThat(num => num != UInt32.max)
+    (signedScriptSig, cltvScriptPubKey, privateKeys) <- ScriptGenerators.signedCLTVScriptSignature(cltvLockTime, txLockTime, sequence)
+  } yield {
+    val (creditingTx, outputIndex) = buildCreditingTransaction(cltvScriptPubKey)
+    val (signedSpendingTx, inputIndex) = buildSpendingTransaction(TransactionConstants.version, creditingTx, signedScriptSig, outputIndex, txLockTime, sequence)
+    val txSigComponent = TransactionSignatureComponent(signedSpendingTx, inputIndex, cltvScriptPubKey, Policy.standardScriptVerifyFlags)
+    (txSigComponent, privateKeys, cltvLockTime)
+  }).suchThat(cltvLockTimesOfSameType)
```

Comment:
Could you move this `.suchThat(...)` to line 165 so we have something like  

`NumberGenerator.uInt32s.suchThat(num => num < txLockTime && cltvLockTimesOfSameType(..))`?